File: CanliYayinSwift/CanliYayinSwift/model/RoomModel.swift
//
//  RoomModel.swift
//  CanliYayinSwift
//
//  Created by Alperen Saraç on 13.05.2026.
//

import Foundation
struct RoomModel {
    let roomId: String
    let title: String
    let broadcasterName: String
    let createdAt: String
    let viewerCount: Int

    init(json: [String: Any]) {
        self.roomId = json["room_id"] as? String ?? ""
        self.title = json["title"] as? String ?? ""
        self.broadcasterName = json["broadcaster_name"] as? String ?? ""
        self.createdAt = json["created_at"] as? String ?? ""
        self.viewerCount = json["viewer_count"] as? Int ?? 0
    }
}

File: flutter_canli_yayin/lib/models/room_model.dart
class RoomModel {
  final String roomId;
  final String title;
  final String broadcasterName;
  final String createdAt;
  final int viewerCount;

  RoomModel({
    required this.roomId,
    required this.title,
    required this.broadcasterName,
    required this.createdAt,
    required this.viewerCount,
  });

  factory RoomModel.fromJson(Map<String, dynamic> json) {
    return RoomModel(
      roomId: json["room_id"] ?? "",
      title: json["title"] ?? "",
      broadcasterName: json["broadcaster_name"] ?? "",
      createdAt: json["created_at"] ?? "",
      viewerCount: json["viewer_count"] ?? 0,
    );
  }
}